pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -645,6 +678,9 @@ func ingressRuleToSDKType(i *infrav1.IngressRule) (res *ec2.IpPermission) {
 		res = &ec2.IpPermission{
 			IpProtocol: aws.String(string(i.Protocol)),
 		}
+	default:
+		scope.Error(fmt.Errorf("invalid protocol '%s'", i.Protocol), "invalid protocol for security group", "protocol", i.Protocol)
```

Comment:
any reason to use scope.Error here? Cant we return error from here directly?

Reply:
Because it would break existing functionality. I just added this to debug when I passed in an incorrect protocol. I don't know if anything else does that, too, maybe, and it would break that flow. But I thought it would be nice to at least surface the problem in the logs. I can try modifying this function to give back and error. Maybe it's okay. :D

Reply:
yeah i understand, just wanted to highlight this from logging guidelines perspective, to be consistent. 🙂

Reply:
It would break up this append:

```go
input.IpPermissions = append(input.IpPermissions, ingressRuleToSDKType(s.scope, &rule))
```

Into an error check for adding a log line. I... don't like that much. :D

Reply:
Hmm got it 👍